### PR TITLE
Add a dispatch to SimpleNewtonRaphson for NNLS and SimpleGaussNewton

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Fixes https://github.com/SciML/SimpleNonlinearSolve.jl/issues/82

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
LBroyden is having issues, @avik-pal can you look into that? https://github.com/SciML/SimpleNonlinearSolve.jl/actions/runs/6690175885/job/18174996133?pr=91#step:6:581

**Comment by avik-pal** [comment]
I will take a look at LBroyden.

File: test/least_squares.jl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+θ_true = [1.0, 0.1, 2.0, 0.5]
+x = [-1.0, -0.5, 0.0, 0.5, 1.0]
+y_target = true_function(x, θ_true)
+
+function loss_function(θ, p)
+    ŷ = true_function(p, θ)
+    return abs2.(ŷ .- y_target)
+end
+
+θ_init = θ_true .+ randn!(similar(θ_true)) * 0.1
```
Comment (ChrisRackauckas) [inline test/least_squares.jl:14]:
```suggestion
θ_init = θ_true .+ 0.1
```

**Comment by avik-pal** [comment]
@ChrisRackauckas this probably needed a version lower bound on SciMLBase

**Comment by avik-pal** [comment]
Also https://github.com/JuliaManifolds/Manifolds.jl/actions/runs/6774649334/job/18412199878#step:9:86 is v"1.98", not even sure how such an old version got installed

**Comment by mateuszbaran** [comment]
I forgot to update BoundaryValueDiffEq compatibility in docs; anyway, this also happens on Julia 1.6: https://github.com/JuliaManifolds/Manifolds.jl/actions/runs/6774649345/job/18412201063 where newer BoundaryValueDiffEq is not available.

**Pull request merged**